clang/test/CIR/CodeGen/complex.c

```diff
@@ -0,0 +1,92 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir-enable -emit-cir %s -o %t.cir
+// RUN: FileCheck --input-file=%t.cir %s
+
+double _Complex float_complex_basic(void) {
+  double _Complex x = 2.0 + 3.0i;
```

Comment:
I'm a bit confused with the approach here, seems a bit less ideal than what LLVM generates: https://godbolt.org/z/3Go78MMh1, looking at the CIR output:
```
    %2 = cir.const(#cir.fp<2.000000e+00> : !cir.double) : !cir.double loc(#loc5)
    %3 = cir.const(#cir.complex<#cir.fp<0.000000e+00> : !cir.double, #cir.fp<3.000000e+00> : !cir.double> : !cir.complex<!cir.double>) : !cir.complex<!cir.double> loc(#loc4)
    %4 = cir.cast(float_to_complex, %2 : !cir.double), !cir.complex<!cir.double> loc(#loc5)
    %5 = cir.binop(add, %4, %3) : !cir.complex<!cir.double> loc(#loc81)
```
Why doesn't `%3` already builds the complex? Seems off that CIRGen first creates `%2`, followed by bitcast + addition.

Reply:
Question still remains, sounds like we could do a better initialization job here. It's also fine if you plan to do this as incremental improvement, but I'm mostly looking to hear more about the plan here.